### PR TITLE
Recipient can reserve a foodbag from index list

### PR DESCRIPTION
**Pull request by FlisAnn**
[Recipient can reserve a foodbag from index list](https://www.pivotaltracker.com/story/show/176800078)
### User Story
```
As an recipient,
In order to make sure I get a foodbag
I want to reserve a foodbag
```
## Changes proposed in this pull request:
* Recipient can reserve a foodbag
* Styling

## What I have learned working on this feature:
* Working with switch
* Creating a put request
* 
## Packages/Gems added
*
## Screenshots (Client)
Simulator Screen Shot - iPhone 12 Pro Max - 2021-02-04 at 22 10 18

### CODE REVIEW

File: components/DisplayFoodBag.jsx [hunk 1]
```diff
@@ -1,43 +1,92 @@
...
+        </Text>
+        <Card.Divider />
+        <Text style={styles.foodbagInfo}>
+          Pickup in the {foodbag.pickuptime}
+        </Text>
+        {!switchValue && foodbag.status === 'available' && (
+          <>
+            <Text style={styles.foodbagReserve}>Reserve this bag</Text>
+            <Switch
+              trackColor={{ false: 'red', true: '#8FBC8F' }}
```
Comment (psnoya) [inline components/DisplayFoodBag.jsx:38]: formatting

File: components/DisplayFoodBagsList.jsx [hunk 2]
```diff
@@ -1,38 +1,54 @@
...
-  )
-}
+    <ScrollView
+      style={{ flex: 1 }}
+      refreshControl={
+        <RefreshControl refreshing={refresh} onRefresh={onRefresh} />
+      }
+    >
+      <View style={styles.container}>
+        <StatusBar style="auto" />
```
Comment (psnoya) [inline components/DisplayFoodBagsList.jsx]: formatting

File: components/RegisterForm.jsx [hunk 3]
```diff
@@ -52,15 +54,16 @@ const RegisterForm = ({ navigation }) => {
       <TextInput
         style={styles.input}
         placeholder='Password confirmation:'
-        secureTextEntry={true}
+        // secureTextEntry={true}
```
Comment (psnoya) [inline components/RegisterForm.jsx]: commented out code + formatting

File: modules/DonorService.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+// const API_URL = 'https://git.heroku.com/weca.git'
+const API_URL = 'http://localhost:3000/api'
+
+const DonorService = {
+  async show(id) {
+    const headers = JSON.parse(await AsyncStorage.getItem('auth-storage'))
+    try {
+      let response = await axios.get(API_URL + `/user/${id}`, {
+        headers: headers,
+      })
```
Comment (psnoya) [inline modules/DonorService.js:13]: formatting

File: state/reducers/rootReducer.js [hunk 5]
```diff
@@ -16,6 +16,16 @@ const rootReducer = (state, action) => {
         ...state,
         registerErrorMessage: action.payload,
       }
+    case 'UPDATE_FOODBAG_INDEX':
+      return {
+        ...state,
+        foodbags: action.payload,
+      }
```
Comment (psnoya) [inline state/reducers/rootReducer.js:23]: formatting

**Review by psnoya — COMMENTED** [review]
Only formatting to do! Really good job

File: components/DisplayFoodBag.jsx [hunk 6]
```diff
@@ -1,43 +1,92 @@
...
+const DisplayFoodBag = ({ foodbag, id }) => {
+  const [switchValue, setSwitchValue] = useState(false)
+
+  useEffect(() => {
+    DonorService.show(id)
+  }, [])
+
+  const toggleSwitch = async foodbag => {
+    setSwitchValue(true)
+    debugger
```
Comment (psnoya) [inline components/DisplayFoodBag.jsx]: remove debugger

File: modules/DonorService.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+      debugger
```
Comment (psnoya) [inline modules/DonorService.js]: remove debugger

File: components/DisplayFoodBag.jsx (hunk 1 above)
Comment (FlisAnn) [reply components/DisplayFoodBag.jsx:38]: yes

File: components/RegisterForm.jsx (hunk 3 above)
Comment (FlisAnn) [reply components/RegisterForm.jsx]: Good catch, putting it back

File: modules/DonorService.js (hunk 4 above)
Comment (FlisAnn) [reply modules/DonorService.js:13]: fixed

File: components/DisplayFoodBag.jsx (hunk 6 above)
Comment (FlisAnn) [reply components/DisplayFoodBag.jsx]: it's gone

File: components/DisplayFoodBagsList.jsx (hunk 2 above)
Comment (FlisAnn) [reply components/DisplayFoodBagsList.jsx]: yes

File: state/reducers/rootReducer.js (hunk 5 above)
Comment (FlisAnn) [reply state/reducers/rootReducer.js:23]: yes

File: modules/DonorService.js (hunk 7 above)
Comment (FlisAnn) [reply modules/DonorService.js]: yeess :)

**Pull request merged**